packages/http/src/validator/utils/spec.ts

```diff
@@ -1,12 +1,27 @@
 import { IHttpOperationResponse } from '@stoplight/types';
+import { fromNullable, Option } from 'fp-ts/lib/Option';
 
 export function findOperationResponse(
   responseSpecs: IHttpOperationResponse[],
   statusCode: number,
-): IHttpOperationResponse | undefined {
+): Option<IHttpOperationResponse> {
   const sortedSpecs = responseSpecs
-    .filter(spec => new RegExp(`^${spec.code.replace(/X/g, '\\d')}$`).test(String(statusCode)))
-    .sort((s1, s2) => s1.code.split('X').length - s2.code.split('X').length);
+    .filter(
+      spec =>
+        new RegExp(`^${spec.code.replace(/X/g, '\\d')}$`).test(String(statusCode)) ||
+        spec.code.toLowerCase() === 'default',
+    )
+    .sort((s1, s2) => {
+      if (s1.code.toLowerCase() === 'default') {
+        return 1;
+      }
 
-  return sortedSpecs[0];
+      if (s2.code.toLowerCase() === 'default') {
+        return -1;
+      }
+
+      return s1.code.split('X').length - s2.code.split('X').length;
+    });
+
+  return fromNullable(sortedSpecs[0]);
```

Comment:
This part is correct! However fp-ts provides some utilities that will help you avoid using `fromNullable` and other stuff.

In particular you could do:

```ts
  return pipe(sortedSpecs, head);
```

but your solution works perfectly anyway.

Reply:
Changed, it looks cooler with that pipe/head approach :)